Arduino_package/hardware/variants/nora_w30/variant.h

```diff
@@ -0,0 +1,107 @@
+/*
+  Copyright (c) 2011 Arduino.  All right reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+/* u-blox NORA-W30 (RTL8720DF) */
+
+#ifndef _VARIANT_ARDUINO_AMEBA_
+#define _VARIANT_ARDUINO_AMEBA_
+
+#include "Arduino.h"
+#include <stdarg.h>
+
+#ifdef __cplusplus
+extern "C"{
+#include "PinNames.h"
+
+#define portOutputRegister(P) ((volatile uint32_t *)(0x48014000 + (P) * 0x400))
+#define portInputRegister(P)  ((volatile uint32_t *)(0x48014050 + (P) * 0x400))
+#define portModeRegister(P)   ((volatile uint32_t *)(0x48014004 + (P) * 0x400))
+
+/*
+ * Wait until enter debug mode
+ *
+ * Check DHCSR(0xE000EDF0) register and hold until bit C_DEBUGEN is set.
+ * Use this function along with J-LINK or other debug tool
+ **/
+extern void wait_for_debug(void);
+}
+#endif
+
+#define MAIN_THREAD_STACK_SIZE                  (4096 * 4)
+
+#define TOTAL_GPIO_PIN_NUM                      (20)
+#define TOTAL_PWM_PIN_NUM                       (12)
+
+/* Digital pin mapping refer to g_APinDescription */
+/*               index into g_APinDescription --v */
+#define D0                                      0 // PB_18

```

Comment:
The define name "D1" is common,  there may have multiple define warning or error.

Reply:
It is still OK to merge in to SDK. Can add some comments above the define

Reply:
Thanks for the quick reply, I could rename them to EVK_Dx or EVK_Ax to avoid this conflict and make sure the river to the pinout of the EVK. @M-ichae-l what do you think?

Reply:
I revised this pull request for NORA-W30 and try to reuse the code from the RTL8720DF target from @daphwl as much as possible. While doing so I found some inconsistencies in the RTL8720DF target comments of variants.cpp as well as defines in boards.txt. 

@daphwl and @M-ichae-l could you please review these changes to the RTL8720DF target as I am new to Realtek / Ameba?

Reply:
Thanks @mazgch ! 

- Under Arduino IDE, "Ax" those analog pin names are OK to use as Arduino boards also use those name. For "Dx", Arduino never use such name that IDE environment never protect such names, therefore it is easier to have multiple define warning or error. In our case, use "EVK_Dx" or "EVK_Ax" is good to solve the issue. I suggest to continue use "Ax", will not have the issue. 
- For "DRTL8722DM" it is link with libs, so we can not change it. I suggest to keep it as usual. 
- For "DBOARD_RTL8720DF" it is use for IC